src/Compilers/Core/Portable/CommandLine/EditorConfig.SectionNameMatching.cs

```diff
@@ -82,13 +97,46 @@ public static string TryCompileSectionNameToRegEx(string sectionName)
                         // Matches any string of characters
                         sb.Append(".*");
                         break;
+                    case TokenKind.OpenCurly:
+                        // This is ambiguous between {num..num} and {item1,item2}
+                        // We need to look ahead to disambiguate. Looking for {num..num}
+                        // is easier because it can't be recursive.
+                        var range = TryParseNumberRange(ref lexer);
+                        if (range is null)
+                        {
+                            // Not a number range. Try a choice expression
+                            if (!TryCompileChoice(ref lexer, sb))
+                            {
+                                return false;
+                            }
```

Comment:
The implicit fall through here to break and then the method return is a bit hard to follow. Consider moving to an explicit return at this location.